corehq/motech/rate_limiter.py

```diff
@@ -55,14 +60,34 @@ def _get_per_user_repeater_wait_milliseconds_rate_definition(domain):
 )
 
 
+def _get_per_user_repeater_attempt_rate_definition(domain):
+
+    return PerUserRateDefinition(
+        per_user_rate_definition=get_dynamic_rate_definition(
+            "repeater_attempts_per_user",
+            default=RateDefinition(
+                per_week=None,
+                per_day=None,
+                per_hour=6000,
+                per_minute=100,
+                per_second=10,
+            ),
```

Comment:
Does this mean that we will throttle after a domain makes more than 10 req/sec or 100 req/min or 6000 req/hour?

Reply:
Yes, if the global wait times are exceeded. These can also be tuned after the fact, and for that reason should be set pretty conservatively, so bumping them higher doesn't sound like a bad idea if that is your concern.

Reply:
No. Its fine. I was just asking for my understanding that it is the way it is supposed to work.

Reply:
Oh but this is a per user definition, so it would actually be much higher. In this case, I wonder if it is better to just use a simple RateDefinition.

Reply:
I think for the goal of this PR, I'm leaning towards switching this to a RateDefinition, otherwise I'm worried large projects could still avoid being rate limited. I view this as the only way we have of limiting the number of workers a project can use at a time, to ensure that other workers remain free to continue processing other records.

Reply:
> I view this as the only way we have of limiting the number of workers a project can use at a time.

I think I might have not understood the changes correctly but where exactly are we limiting the number of workers?

Reply:
Yeah sorry I mentioned this the comment on the feature flag description as well. We aren't explicitly limiting the number of workers, but conceptually are (I think). The idea is that by limiting the number of records a domain can attempt to forward in a given time window, we are limiting the number of workers they will use in that time window too. If a domain has records that take ~a minute to send to an endpoint, and is able to attempt records for all 100 celery workers (let's pretend 100 is the total concurrency), for that minute, all celery workers will be busy attempting to forward records for that domain. If we instead say a domain can only attempt 10 forwards a minute, that domain is only able to use 10% of our total concurrency/number of workers.

The numbers are obviously way off, but that is the idea.